pkg/chains/signing.go

```diff
@@ -58,21 +62,45 @@ type TaskRunSigner struct {
 }
 
 // IsSigned determines whether a TaskRun has already been signed.
-func IsSigned(tr *v1beta1.TaskRun) bool {
+func IsSigned(tr *v1beta1.TaskRun) (bool, string) {
 	signature, ok := tr.ObjectMeta.Annotations[ChainsAnnotation]
 	if !ok {
```

Comment:
I don't think we need this check -- if it doesn't exist then signature will be an empty string and we'll get `false` back anyway

Reply:
Addressed it.

Reply:
In that case let's create a `State` type and instead have a function that returns the state --

```go
type State int

const (
    Passed State = iota
    Failed
    Retrying
)

// returns the appropriate State
func State(tr *v1beta1.TaskRun) State
```